torch/_ops.py

```diff
@@ -99,39 +99,59 @@ def __getattr__(self, key):
         if key == '__file__':
             return 'torch.ops'
 
+        # ensure that query for dunder attributes that does not exist on
+        # opoverloadpacket but instead exists on the self._op object does not unnecessarily call
+        # `_get_operation_overload` (which is an expensive operation).
+        # This is done to prevent any potential slowdown. This list can be extended
+        # if there exists other attributes like `__name__` that only exist on self._op and not on the
+        # opoverloadpacket.
+        try:
+            if key.startswith('__') and key.endswith('__'):
+                return getattr(self._op, key)
+        except AttributeError:
+            # for consistency because it seems weird to
+            # throw an attribute error with a message containing
+            # an object name different from the one the attribute
+            # query was performed on.
+            raise AttributeError("'{}' object has no attribute '{}'".format(str(self), key)) from None
```

Comment:
Why do you set "from None" ?

Reply:
I had nested try except before so needed it but don't need it anymore

Reply:
without from None:
```
>>> torch.ops.aten.add.__bla
Traceback (most recent call last):
  File "/fsx/users/chourdiaanjali/pytorch/torch/_ops.py", line 111, in __getattr__
    return getattr(self._op, key)
AttributeError: 'builtin_function_or_method' object has no attribute '__bla'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/fsx/users/chourdiaanjali/pytorch/torch/_ops.py", line 117, in __getattr__
    raise AttributeError("'{}' object has no attribute '{}'".format(str(self._op), key))
AttributeError: '<built-in method add of PyCapsule object at 0x7f4b01952060>' object has no attribute '__bla'
```

with it:
```
>>> torch.ops.aten.add.__bla
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/fsx/users/chourdiaanjali/pytorch/torch/_ops.py", line 117, in __getattr__
    raise AttributeError("'{}' object has no attribute '{}'".format(str(self._op), key)) from None
AttributeError: '<built-in method add of PyCapsule object at 0x7fc270924060>' object has no attribute '__bla'
```

Reply:
come to think of it, maybe I don't need a try except here

Reply:
If we make the re-throw message better, I guess we can keep the from None.